src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const sep = separator;
+
+    if (this.length === 0) {
+      return result;
+    }
```

Comment:
There is no need to check it. If you will have an empty array your cycle will not do anything and the function will return the result immediately.